daemon/create.go

```diff
@@ -151,5 +153,9 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 	if err != nil {
 		return nil, err
 	}
+
+	if (driverName != "" && v.DriverName() != driverName) || (driverName == "" && v.DriverName() != volume.DefaultDriverName) {
+		return nil, fmt.Errorf("Conflict. Cannot create volume '%s' using the '%s' driver because the volume '%s' already exists and the driver name is '%s'.", name, driverName, name, v.DriverName())
```

Comment:
@azurezk do you mind to move this error to our new `errors` package?

See examples in: https://github.com/docker/docker/blob/master/errors/daemon.go